kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSPropertyDeclarationImpl.kt

```diff
@@ -96,8 +112,9 @@ class KSPropertyDeclarationImpl(private val ktPropertySymbol: KtPropertySymbol)
 
     override val parent: KSNode? by lazy {
         analyzeWithSymbolAsContext(ktPropertySymbol) {
-            ktPropertySymbol.getContainingSymbol()?.let { KSClassDeclarationImpl(it as KtNamedClassOrObjectSymbol) }
-                ?: ktPropertySymbol.toContainingFile()
+            ktPropertySymbol.getContainingSymbol()?.let {
+                KSClassDeclarationImpl.getCached(it as KtNamedClassOrObjectSymbol)
+            } ?: ktPropertySymbol.toContainingFile()
```

Comment:
Just curious; when it's not a member property, I guess you assumed a top-level property and retrieved the containing file. Shouldn't the parent be a facade class?

Reply:
Facade class makes more sense at file compiled class, but from source code they appears to be contained directly in the file, therefore KSP models in this way.